CMakeLists.txt

```diff
@@ -37,12 +37,17 @@ set(SHARED_UTIL_SRC_FOLDER ${CMAKE_CURRENT_LIST_DIR}/src CACHE INTERNAL "this is
 set(SHARED_UTIL_ADAPTER_FOLDER "${CMAKE_CURRENT_LIST_DIR}/adapters" CACHE INTERNAL "this is where the adapters live" FORCE)
 
 include_directories(${SHARED_UTIL_INC_FOLDER})
-
+get_property(dirs DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR} PROPERTY INCLUDE_DIRECTORIES)
+foreach(dir ${dirs})
+  message(STATUS "dir='${dir}'")
+endforeach()
 CHECK_INCLUDE_FILES(stdint.h HAVE_STDINT_H)
 CHECK_INCLUDE_FILES(stdbool.h HAVE_STDBOOL_H)
 
-if ((NOT HAVE_STDINT_H) OR (NOT HAVE_STDBOOL_H))
+if (WINDOWS)
```

Comment:
I was cross compiling for a linux-armhf (Debian 8) device.
Due to bad linker flags, cmake claimed that it can not find stdint.h nor stdbool.h.
This had caused, that cmake accidentally adds the windowsce directory to the include path.
Which made it even more difficult, to see what is really going wrong.